thunderfish/sortfishtype.py

```diff
@@ -0,0 +1,177 @@
+     following trough, relative to the time between consecutive peaks.
+
+    :param data: (1-D array) The data to be analyzed (Usually the best window already)
+    :param samplerate: (float). Sampling rate of the data in Hz
+    :param win_shift: (float). Time shift in seconds between windows.
+    :param pulse_thres: (float): a positive number setting the minimum distance between peaks and troughs
+    :param plot_data_func: Function for plotting the data with detected peaks and troughs, an inset and the distribution
+    of peak-ratios.
+        plot_data_func(data, rate, peak_idx, trough_idx, idx0, idx1,
+                       win_start_times, cv_interv, mean_ampl, cv_ampl, clipped_frac, cost,
+                       thresh, valid_wins, **kwargs)
+        :param data (array): the raw data.
+        :param rate (float): the sampling rate of the data.
+        :param peak_idx (array): indices into raw data indicating detected peaks.
+        :param trough_idx (array): indices into raw data indicating detected troughs.
+    :return: suggestion: (str). Returns the suggested fish-type ("pulse"/"wave").
+    :return: peak_ratio: (float). Returns a float between 0. and 1. which gives the proportion of peak-2-trough,
+                            from peak-2-peak time distance. (Wave-fishes should have larger values than pulse-fishes)
+    """
+
+    print('\nAnalyzing Fish-Type...')
+
+    # threshold for peak detection:
+    threshold = np.zeros(len(data))
+    win_shift_indices = int(win_shift * samplerate)
+
+    for inx0 in xrange(0, len(data) - win_shift_indices / 2, win_shift_indices):
+        inx1 = inx0 + win_shift_indices
+        threshold[inx0:inx1] = np.percentile(data[inx0:inx1], 99)
+
+    # detect large peaks and troughs:
+    peak_idx, trough_idx = pkd.detect_peaks(data, threshold)
+    peak_idx, trough_idx = pkd.trim_closest(peak_idx, trough_idx)
+
+    # get times of peaks and troughs
+    pk_times = peak_idx/samplerate
+    tr_times = trough_idx/samplerate
+
+    pk_2_pk = np.diff(pk_times)
+    pk_2_tr = np.abs(pk_times - tr_times)[:-1]
+
+    # get the proportion of peak-2-trough, from peak-2-peak time distance
+    r_tr = pk_2_tr / pk_2_pk
+    r_tr[r_tr > 0.5] = 1 - r_tr[r_tr > 0.5]
+    # ToDo: Some values are not between 0. and 1. as they should! Need to find a solution!
+
+    peak_ratio = np.median(r_tr)
+    suggestion = 'pulse' if peak_ratio < pulse_thres else 'wave'
+
+    if plot_data_func:
+        plot_data_func(data, samplerate, peak_idx, trough_idx, threshold, pulse_thres, suggestion, r_tr)
+
+    print('\nFish-type is %s. r-value = %.3f' % (suggestion, peak_ratio))
+    return suggestion, peak_ratio
+
+
+def plot_type_detection(data, rate, peak_idx, trough_idx, peakdet_th, r_value_th, type_suggestion, r_vals):
+    """ Plots the best-window, a zoomed index of it and the r-values that determine whether fish is pulse or wave-type.
```

Comment:
Ok, I accept the pull request. So that Till can proceed. But we will continue to work on the algorithms.